tests/backup/backup_upgrade_test.go

```diff
@@ -204,3 +206,419 @@ var _ = Describe("{StorkUpgradeWithBackup}", func() {
+			}
+			_ = TaskHandler(Inst().AppList, createSingleNSBackupTask, Parallel)
+		})
+		Step("Create schedule backup of all namespaces", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			allNamespaces = make([]string, 0)
+			for _, appName := range Inst().AppList {
+				allNamespaces = append(allNamespaces, dstClusterAppNamespaces[appName]...)
+			}
+			allNSScheduleName = fmt.Sprintf("%s-all-ns-schedule-%v", BackupNamePrefix, time.Now().Unix())
+			labelSelectors := make(map[string]string)
+			log.InfoD("Creating schedule backup with schedule [%s] of all namespaces of destination cluster [%s]", allNSScheduleName, allNamespaces)
+			err = CreateScheduleBackup(allNSScheduleName, destinationClusterName, backupLocationName, backupLocationUid, allNamespaces,
+				labelSelectors, orgID, "", "", "", "", schedulePolicyName, schedulePolicyUid, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of schedule backup with schedule [%s]", allNSScheduleName))
+			firstScheduleBackupName, err := GetFirstScheduleBackupName(ctx, allNSScheduleName, orgID)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching name of the first schedule backup with schedule [%s]", allNSScheduleName))
+			log.Infof("First schedule backup name: [%s]", firstScheduleBackupName)
+		})
+		Step("Upgrading px-backup", func() {
+			latestPxBackupVersionFromEnv := os.Getenv("LATEST_PXBACKUP_VERSION")
+			if latestPxBackupVersionFromEnv == "" {
+				latestPxBackupVersionFromEnv = latestPxBackupVersion
+			}
+			log.InfoD("Upgrading px-backup to latest version [%s]", latestPxBackupVersionFromEnv)
+			err := UpgradePxBackup(latestPxBackupVersionFromEnv)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying completion of px-backup upgrade to latest version [%s]", latestPxBackupVersionFromEnv))
+		})
+		Step("Restore full backups created with and without pre- and post-exec rules", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			log.InfoD("Restoring full backups [%s] created with rules", backupWithRuleNames)
+			for _, backupName := range backupWithRuleNames {
+				namespaceMapping := make(map[string]string, 0)
+				storageClassMapping := make(map[string]string, 0)
+				restoreName := fmt.Sprintf("%s-%s-%v", "test-restore", backupName, time.Now().Unix())
+				log.InfoD("Restoring backup [%s] in cluster [%s] with restore [%s]", backupName, destinationClusterName, restoreName)
+				err = CreateRestore(restoreName, backupName, namespaceMapping, destinationClusterName, orgID, ctx, storageClassMapping)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying restoration [%s] of backup [%s] in cluster [%s]", restoreName, backupName, destinationClusterName))
+				restoreNames = append(restoreNames, restoreName)
+			}
+			log.InfoD("Restoring full backups [%s] created without rules", backupWithoutRuleNames)
```

Comment:
Same question here.

Reply:
Done. Added a helper function IsFullBackup to check if the backup is full backup or not.